client/state/index.js

```diff
@@ -149,13 +149,17 @@ export function createReduxStore( initialState = {} ) {
 
 	const middlewares = [
 		thunkMiddleware,
+		// We need the data layer middleware to be used as early
+		// as possible, before any side effects.
+		// This avoids, for example, side effect actions like analytics
+		// being dispatched more than once.
+		require( './data-layer/wpcom-api-middleware.js' ).default,
```

Comment:
would hate to test that specifically @gwwar because it's just another test testing that the code we wrote is the code we wrote. much better would be a behavioral test which asserts that a desired behavior is produced.

```
it( 'should not process network responses outside of the http middleware', () => {
	…
} );
```